Extensions/rnaoptions.hh

```diff
@@ -65,6 +71,38 @@ class Opts {
   private:
     Opts(const Opts&);
     Opts &operator=(const Opts&);
+
+	int parse_checkpointing_interval(const std::string &interval) {
+      // parse the user-specified checkpointing interval
+      std::stringstream tmp_interval(interval);
+      std::string val;
+      std::vector<int> interval_vals;
+
+      try {
+        // parse the checkpointing interval the user specified
+        while (std::getline(tmp_interval, val, ':')) {
+          // split the interval string at ':' and store values
+          interval_vals.push_back(std::stoi(val));
+        }
+
+        if (interval_vals.size() != 4) {
+          throw std::exception();
+        }
+      } catch (const std::exception &e) {
+        throw OptException("Invalid interval format! "
+                           "Must be d:h:m:s (e.g. 0:0:1:0).");
+      }
+
+      // calculate the interval length (in seconds)
+      int cp_interval = interval_vals[0] * 86400 + interval_vals[1] * 3600 +
+                        interval_vals[2] * 60 + interval_vals[3];
+
+      if (cp_interval <= 0) {
+        throw OptException("Interval cannot be <= 0 (is " +
+                           std::to_string(cp_interval) + ").");
+      }
+      return cp_interval;
+    }
```

Comment:
would it make sense to move this function into the https://github.com/jlab/gapc/blob/f6d487ade9a6e4ea08f825e1e2af8a4eafedf068/src/checkpoint.hh file to avoid the code duplication here and in the according generic options file in the gapc repo?

Reply:
That would definitely make sense. I will move this function to that file.